Add support for ARM64 darwin

hjmallon opened the pull request:
* Rename linux specific functions and file.
* Use sysctlbyname to get ARM64 CRC32 instruction availability.
* Tested on ARM64 macOS machine. This might also work on iOS (untested!)

pwnall reviewed (approved):
This LGTM. I made a small comment.

Would you like to add yourself to AUTHORS here, before I merge?

hjmallon commented on src/crc32c_arm64_check.h:
```diff
@@ -25,17 +24,27 @@ extern "C" unsigned long getauxval(unsigned long type) __attribute__((weak));
+#elif defined(__APPLE__)
+  int val = 0;
+  size_t len = sizeof(val);
+  return sysctlbyname("hw.optional.armv8_crc32", &val, &len, nullptr, 0) == 0 && val != 0;
```
I have wrapped it

hjmallon commented:
I have added myself to AUTHORS and addressed the comment. 

Further to this change I have had to vendor this repo into my own code as building a universal binary from the current code does not work. Just as an FYI for anyone looking at this and having trouble. :)
